File: Data-Structure/Session-Overview/StackDataStructure/Stack.java
package StackDataStructure;

/**
 * @author Niraj Gupta
 * 
 * Interface to let the user know what operation he can perform on stack data structure.
 * @param <DataType> : User decide what kind of data is required in stack. 
 */
public interface Stack<DataType>{
	/**
	 * Method to get the top element of stack
	 * @return : The top of the stack
	 * @throws Exception : If the stack is empty than there is no element at the top of the stack.
	 */
	DataType getTopElement() throws Exception;
	
	/**
	 * Method to push the element on the top of stack
	 * @param element : The data which is to be inserted at the top of stack.
	 */
	void push(DataType element);
	
	/**
	 * Method to delete or remove the top element of the stack.
	 * @return : The top element of the stack which is removed from the stack.
	 */
	DataType pop();
	
	/**
	 * Method to check if the stack is empty or not.
	 * @return : True if the stack is empty, false otherwise.
	 */
	boolean isEmpty();
	
	/**
	 * Method to get the size of the stack.
	 * @return : The total number of element present in the stack.
	 */
	int size();
}

File: UI-Design-And-Development/Session-8/javascript/stack.js
/**
Stack Class or Constructor via function
*/
function Stack() {
    this.size = 0;
    this.storage = {}; 
}

/**
Method to add or push data into the stack
*/
Stack.prototype.push = function(data) {
    var size = ++this.size; 
    this.storage[size] = data;
};

/**
Method to delete or pop data from the stack
*/
Stack.prototype.pop = function() {
    var size = this.size, deleteData;
    if (size) {
       deleteData = this.storage[size];
       delete this.storage[size];
       this.size--;
       return deleteData; 
    }
};